test/functional/fixtures/tty-test.c

```diff
@@ -150,7 +150,12 @@ int main(int argc, char **argv)
   }
 
   if (argc > 1) {
-    int count = atoi(argv[1]);
+    char *endptr;
+    errno = 0;
+    int count = (int)strtol(argv[1], &endptr, 10);
+    if (errno != 0) {
+      abort();
+    }
```

Comment:
Would something like `count = min(count, some_sensible_number);` not be enough to make sure the loop range is safe?

Reply:
That's what I did at first, but then wasn't sure if there was some detail in the `atoi` implementation or spec that may break on unexpected input.

Reply:
Man page says the behavior is the same as `strtol(argv[1],NULL,10);` except that errors are not detected.

I would assume Covertiy complains about a size unchecked upper bound for the loop that could be too big (Did the programmer really consider what an upper bound of 10^9 would do? What about a negative value?).

Reply:
I added a bounds check.